airbrakes/data_handling/logged_data_packet.py

```diff
@@ -59,30 +58,101 @@ class LoggedDataPacket(msgspec.Struct):
     estGravityVectorZ: float | None = None
 
     # Processed Data Packet Fields
-    avg_acceleration: tuple[float, float, float] | None = None
     current_altitude: float | None = None
     speed: float | None = None
     # Not logging maxes because they are easily found
 
     def set_imu_data_packet_attributes(self, imu_data_packet: IMUDataPacket) -> None:
         """
-        Sets the attributes of the data packet corresponding to the IMU data packet. Performs
-        some processing to round the float values to 8 decimal places.
+        Sets the attributes of the data packet corresponding to the IMU data packet. Rounds the
+        float values to 8 decimal places, if the value is a float.
+
+        This function could be a lot cleaner and just use getattr() and setattr(), but that is
+        slower and takes up 15% of the main loop execution time.
+
+        :param imu_data_packet: The IMU data packet to set the attributes from.
         """
-        for key in imu_data_packet.__struct_fields__:
-            if hasattr(self, key):
-                value = getattr(imu_data_packet, key)
-                # Only logs the 8 decimal places as there is already so much noise in the data
-                if isinstance(value, float):
-                    value = round(value, DATA_PACKET_DECIMAL_PLACES)
-                setattr(self, key, value)
-            else:
-                raise AttributeError(f"{key} is not a valid attribute")
+        idp = imu_data_packet
```

Comment:
same here

Reply:
@JacksonElia ?

Reply:
I still think it shouldn't be abbreviated